File: datasets/Part 2 - Regression/Section 4 - Simple Linear Regression/simple_linear_regression_sbn.R
# Regresion lineal simple
# Importar dataset
dataset <- read.csv('Salary_Data.csv')
#dataset = dataset[, 2:3]

# Dividir el dataset en entrenamiento y prueba
#install.packages('caTools')
library(caTools)
#set.seed(123) SOLO ES PARA QUE SIEMPRE SELECCIONE LOS MISMOS ELEMENTOS
split <- sample.split(dataset$Salary, SplitRatio = 2/3)
trainning_set <- subset(dataset, split == TRUE)
testing_set <- subset(dataset, split == FALSE)

# PARA REGRESION LINEAL SIMPLE NO SE REQUIERE ESCALAR DATOS
# Escalado de variables
# trainning_set[, 2:3] <- scale(trainning_set[, 2:3])
# testing_set[, 2:3] <- scale(testing_set[, 2:3])

# Crear modelo de regresion lineal simple con los datos de entrenamiento
regression <- lm(formula = Salary ~ YearsExperience,
                 data = trainning_set)

#summary(regression)

# Predecir el conjunto de prueba
# PARA USAR PREDICT LAS COLUMNAS DEL DATASET Y DE LOS NUEVOS DATOS DEBEN LLAMARSE IGUAL
y_pred <- predict(regression,
                  newdata = testing_set)

# Visualizar los resultados de entrenamiento
library(ggplot2)
ggplot() +
  geom_point(aes(x = trainning_set$YearsExperience,
                 y = trainning_set$Salary),
             colour = 'red') +
  geom_line(aes(x = trainning_set$YearsExperience,
                y = predict(regression,
                            newdata = trainning_set)),
            colour = 'blue') +
  ggtitle('Sueldo vs Años de experiencia [Conjunto de Entrenamiento]') +
  xlab('Años de experiencia') +
  ylab('Sueldo (USD $)')

# Visualizar los resultados de prueba
library(ggplot2)
ggplot() +
  geom_point(aes(x = testing_set$YearsExperience,
                 y = testing_set$Salary),
             colour = 'red') +
   geom_line(aes(x = trainning_set$YearsExperience,
                 y = predict(regression,
                             newdata = trainning_set)),
             colour = 'blue') +
  # geom_line(aes(x = testing_set$YearsExperience,
  #               y = y_pred),
  #           colour = 'blue') +
  ggtitle('Sueldo vs Años de experiencia [Conjunto de Test]') +
  xlab('Años de experiencia') +
  ylab('Sueldo (USD $)')

File: datasets/Part 2 - Regression/Section 4 - Simple Linear Regression/simple_linear_regression_sbn.py
# -*- coding: utf-8 -*-
"""
Created on Thu Jan  9 12:15:16 2020

@author: jguerrero
"""

# Regresion lineal simple
# Como importar las librerias
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importar el dataset
dataset = pd.read_csv('Salary_Data.csv')

X = dataset.iloc[:,:-1].values
y = dataset.iloc[:, 1].values

# Dividir el dataset en entrenamiento y prueba
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3)#, random_state = 0) SOLO ES PARA QUE SIEMPRE SELECCIONE LOS MISMOS ELEMENTOS

# PARA REGRESION LINEAL SIMPLE NO SE REQUIERE ESCALAR DATOS
# Escalado de variables
# Estandarizacion
'''from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
# Escalar los datos de train y test con la misma formula
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)'''

# Crear modelo de regresion lineal simple con los datos de entrenamiento
from sklearn.linear_model import LinearRegression
regression = LinearRegression()
regression.fit(X_train, y_train)

# Predecir el conjunto de prueba
y_pred = regression.predict(X_test)

# Visualizar los resultados de entrenamiento
plt.scatter(X_train, y_train, color = 'red')
plt.plot(X_train, regression.predict(X_train), color = 'blue')
plt.title('Sueldo vs Años de experiencia [Conjunto de Entrenamiento]')
plt.xlabel('Años de experiencia')
plt.ylabel('Sueldo (USD $)')
plt.show()

# Visualizar los resultados de prueba
plt.scatter(X_test, y_test, color = 'red')
plt.plot(X_train, regression.predict(X_train), color = 'blue')
#plt.plot(X_test, y_pred, color = 'blue')
plt.title('Sueldo vs Años de experiencia [Conjunto de Test]')
plt.xlabel('Años de experiencia')
plt.ylabel('Sueldo (USD $)')
plt.show()